client/src/context/useBoardContext.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState, useContext, createContext, FunctionComponent } from 'react';
+import { IBoardData, INewCard } from '../interface/Board';
+import { IMoveAction } from '../interface/BoardActions';
+import reducer from '../utils/reducer';
+import { useImmerReducer } from 'use-immer';
+import { useAuth } from './useAuthContext';
+import createBoard from '../helpers/APICalls/createBoard';
+import { useEffect } from 'react';
+
+interface IBoardContext {
+  state: IBoardData | undefined;
+  dispatch: (arg: INewCard | IMoveAction | string | { title: string; side: string }) => void;
+  setBoardIndex: (arg: number) => void;
+}
+
+export const BoardContext = createContext<IBoardContext>({
+  state: undefined,
+  dispatch: () => null,
+  setBoardIndex: () => null,
+});
+
+export const BoardProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const { loggedInUser } = useAuth();
+  const [boardIndex, setBoardIndex] = useState(0);
+  const [selectedBoard, setSelectedBoard] = useState<IBoardData>(
+    loggedInUser?.boards[boardIndex] ?? ({} as IBoardData),
+  );
+
+  useEffect(() => {
+    const checkForBoard = async () => {
+      if (loggedInUser && !loggedInUser.boards.length) {
+        // if the user does not have any boards yet, create one
+        const newBoard = await createBoard('New board', loggedInUser.id);
+        if (newBoard.success) {
+          loggedInUser.boards.push(newBoard.success.board);
```

Comment:
This is not a good UX, what if the user doesn't want any boards? Automatically just creating boards for a user isn't really necessary.

Reply:
Rather, we should display a different UI for empty boards such as "Get started by creating a new board"